bin/fit_posparams

```diff
@@ -131,316 +86,33 @@ for file in infiles:
-    return table['CTRL_ENABLED'] == False
-
-def _no_move_performed(table):
-    '''Returns array of bools whether a move was performed in each row of table.
-    Always assumed True for first row.'''
-    dummy = table['TOTAL_MOVE_SEQUENCES'][0] - 1
-    moveseq_diff = _np_diff_with_prepend(table['TOTAL_MOVE_SEQUENCES'], prepend=dummy)
-    return moveseq_diff == 0
-
-def _no_cruise_performed(table):
-    '''Returns array of bools whether a cruise move was performed in each row
-    of table. Always assumed True for first row.'''
-    dummyT = table['TOTAL_MOVE_SEQUENCES'][0] - 1
-    dummyP = table['TOTAL_MOVE_SEQUENCES'][0] - 1
-    cruiseT_diff = _np_diff_with_prepend(table['TOTAL_CRUISE_MOVES_T'], prepend=dummyT)
-    cruiseP_diff = _np_diff_with_prepend(table['TOTAL_CRUISE_MOVES_P'], prepend=dummyP)
-    return cruiseT_diff + cruiseP_diff == 0 # these diffs are always >= 0 for sorted table input
-
-# which filter functions to apply
-bad_row_funcs = [_ctrl_not_enabled, _no_move_performed] # JHS 2020-04-16 Intentionally not including no_cruise_performed()
-
-# remove bad rows
-for posid, table in tables.items():
-    initial_len = len(table)
-    for func in bad_row_funcs:
-        bad_rows = func(table)
-        table.remove_rows(bad_rows)
-        if len(table) == 0:
-            break
-    final_len = len(table)
-    printlog(f'{posid}: dropped {initial_len - final_len} of {initial_len} non-conforming data rows')
-_drop_empty_tables()
-
-# add column for seconds-since-epoch version of date to data frames
-for posid, table in tables.items():
-    dates = table['DATE']
-    table['DATE_SEC'] = Time(dates, format='iso').unix
-    printlog(f'{posid}: generated seconds-since-epoch column (\'DATE_SEC\')')
-
-# data selectors
-def select_by_index(table, start=0, final=-1):
-    '''Returns a subset of data formatted for the best-fitting function.'''
-    data = {}
-    subtable = table[start:final]
-    data['posintT'] = subtable['POS_T'].tolist()
-    data['posintP'] = subtable['POS_P'].tolist()
-    data['ptlX'] = subtable['X_PTL'].tolist()
-    data['ptlY'] = subtable['Y_PTL'].tolist()
-    return data, subtable
-
-def _row_idx_for_time(presorted_table, t):
-    '''Returns index of first row in table which matches the argued time t.
-    Value t should be in same scale as 'DATE_SEC' column. I.e. the date in
-    seconds-since-epoch. If t is outside the time range of the table, the
-    function will return either index 0 or max index.
     
-    Note: For speed, this function assumes the provided table has already been
-    pre-sorted, ascending by time!
-    '''
-    if t < presorted_table['DATE_SEC'][0]:
-        return 0
-    if t > presorted_table['DATE_SEC'][-1]:
-        return len(presorted_table) - 1
-    return int(np.argwhere(table['DATE_SEC'] >= t)[0])
+# verbosity control
+if args.verbose:
+    printf = print
+else:
+    def printf(*args, **kwargs):
+        pass
```

Comment:
Minor: in general we prefer to use `desimeter.log.get_logger(level)` to get a logger, and then use `log.debug()`, `log.info()`, `log.warn()`, `log.error()`, `log.critical()` throughout the code instead of print statements.  The logging outputs are uglier to read, but include filenames and line numbers which are handy for debugging, and you can change the verbosity when running the code via $DESI_LOGLEVEL, or passing level name into get_logger().

This isn't a blocking factor for this PR, and it's ok to sometimes have exceptions for not using the logger, but in the future consider using the logger instead of print/printf.

Reply:
Sounds good. Yes I will start using these in the future